specification/eventgrid/typespec/main.tsp

```diff
@@ -0,0 +1,230 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-autorest";
+import "@azure-tools/typespec-azure-core";
+
+enum ServiceApiVersions {
+  v2022_05_01: "2023-06-01-preview"
+}
+
+@useAuth(
+  ApiKeyAuth<ApiKeyLocation.header, "aeg-sas-key"> | OAuth2Auth<[{
```

Comment:
While testing the service, what we see is that the only autg support is:
Authorisation: SharedAccessKey <key>